akka-http2-support/src/main/scala/akka/http/impl/http2/Protocol.scala

```diff
@@ -0,0 +1,148 @@
+     *    directional: it applies to the number of streams that the sender
+     *    permits the receiver to create.  Initially, there is no limit to
+     *    this value.  It is recommended that this value be no smaller than
+     *    100, so as to not unnecessarily limit parallelism.
+     *
+     *    A value of 0 for SETTINGS_MAX_CONCURRENT_STREAMS SHOULD NOT be
+     *    treated as special by endpoints.  A zero value does prevent the
+     *    creation of new streams; however, this can also happen for any
+     *    limit that is exhausted with active streams.  Servers SHOULD only
+     *    set a zero value for short durations; if a server does not wish to
+     *    accept requests, closing the connection is more appropriate.
+     */
+    case object SETTINGS_MAX_CONCURRENT_STREAMS extends SettingIdentifier(0x3)
+
+    /**
+     * SETTINGS_INITIAL_WINDOW_SIZE (0x4):  Indicates the sender's initial
+     *    window size (in octets) for stream-level flow control.  The
+     *    initial value is 2^16-1 (65,535) octets.
+     *
+     *    This setting affects the window size of all streams (see
+     *    Section 6.9.2).
+     *
+     *    Values above the maximum flow-control window size of 2^31-1 MUST
+     *    be treated as a connection error (Section 5.4.1) of type
+     *    FLOW_CONTROL_ERROR.
+     */
+    case object SETTINGS_INITIAL_WINDOW_SIZE extends SettingIdentifier(0x4)
+
+    /**
+     * SETTINGS_MAX_FRAME_SIZE (0x5):  Indicates the size of the largest
+     *    frame payload that the sender is willing to receive, in octets.
+     *
+     *    The initial value is 2^14 (16,384) octets.  The value advertised
+     *    by an endpoint MUST be between this initial value and the maximum
+     *    allowed frame size (2^24-1 or 16,777,215 octets), inclusive.
+     *    Values outside this range MUST be treated as a connection error
+     *    (Section 5.4.1) of type PROTOCOL_ERROR.
+     */
+    case object SETTINGS_MAX_FRAME_SIZE extends SettingIdentifier(0x5)
+
+    /**
+     * SETTINGS_MAX_HEADER_LIST_SIZE (0x6):  This advisory setting informs a
+     *    peer of the maximum size of header list that the sender is
+     *    prepared to accept, in octets.  The value is based on the
+     *    uncompressed size of header fields, including the length of the
+     *    name and value in octets plus an overhead of 32 octets for each
+     *    header field.
+     *
+     *    For any given request, a lower limit than what is advertised MAY
+     *    be enforced.  The initial value of this setting is unlimited.
+     */
+    case object SETTINGS_MAX_HEADER_LIST_SIZE extends SettingIdentifier(0x6)
+
+    val All =
+      Seq(
+        SETTINGS_HEADER_TABLE_SIZE,
+        SETTINGS_ENABLE_PUSH,
+        SETTINGS_MAX_CONCURRENT_STREAMS,
+        SETTINGS_INITIAL_WINDOW_SIZE,
+        SETTINGS_MAX_FRAME_SIZE,
+        SETTINGS_MAX_HEADER_LIST_SIZE)
+        .map(s ⇒ s.id → s)
+        .toMap
```

Comment:
leave a TODO that this could be array instead of Map? Indexing right into array is ❤️